sdk/dotnet/Pulumi.Automation/Types/ExportTypes.cs

```diff
@@ -0,0 +1,252 @@
+// Copyright 2016-2021, Pulumi Corporation
+// NOTE: The classes in this file are intended to align with the serialized
+// JSON types defined and versioned in sdk/go/common/apitype/core.go
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+// TODO(vipentti): Select proper namespace
+namespace Pulumi.Automation.Types
+{
+    // TODO(vipentti): internal -> public
+    // TODO(vipentti): Split into file per class
+    // TODO(vipentti): Separate JSON models & public types?
+
+    internal class VersionedDeployment
+    {
+        public int? Version { get; set; }
+
+        public DeploymentV3 Deployment { get; set; } = null!;
+    }
```

Comment:
This type is basically `UntypedDeployment` with a single specified deployment type